### PR TITLE
Cache packages from install mirror

### PR DESCRIPTION
**Pull request by cdonati**

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1406/

File: debian/appscale_install_functions.sh [hunk 1]
```diff
@@ -313,35 +330,34 @@ postinstallnginx()
...
-      $APPSCALE_PACKAGE_MIRROR/solr-${SOLR_VER}.tgz
-    tar xzf solr-${SOLR_VER}.tgz
-    mv -v solr-${SOLR_VER} solr
-    rm -fv solr-${SOLR_VER}.tgz
+    SOLR_DIR="${APPSCALE_HOME}/SearchService/solr"
+    mkdir -p ${SOLR_DIR}
+    rm -rf "${SOLR_DIR}/solr"
+
+    SOLR_PACKAGE="solr-${SOLR_VER}.tgz"
+    cachepackage ${SOLR_PACKAGE} "a24f73f70e3fcf6aa8fda67444981f78"
```
Comment (menivaitsi) [inline debian/appscale_install_functions.sh]: Can we have the MD5 in a var?

Comment (menivaitsi) [inline debian/appscale_install_functions.sh]: If you think it's more readable this way, that's fine too.

File: debian/appscale_install_functions.sh [hunk 2]
```diff
@@ -313,35 +330,34 @@ postinstallnginx()
...
-    tar xzf apache-cassandra-${CASSANDRA_VER}-bin.tar.gz
-    mv -v apache-cassandra-${CASSANDRA_VER} cassandra
-    rm -fv apache-cassandra-${CASSANDRA_VER}-bin.tar.gz
-    cd cassandra
-    chmod -v +x bin/cassandra
-    cp -v ${APPSCALE_HOME}/AppDB/cassandra/templates/cassandra.in.sh ${APPSCALE_HOME}/AppDB/cassandra/cassandra/bin
-    cp -v ${APPSCALE_HOME}/AppDB/cassandra/templates/cassandra-env.sh ${APPSCALE_HOME}/AppDB/cassandra/cassandra/conf
+
+    CASSANDRA_PACKAGE="apache-cassandra-${CASSANDRA_VER}-bin.tar.gz"
+    cachepackage ${CASSANDRA_PACKAGE} "1894c5103d12a2be14a2c44bfa2363cc"
```
Comment (menivaitsi) [inline debian/appscale_install_functions.sh]: Same here.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Create-Eucalyptus-Image/2237/

**Pull request merged**